ibc-derive/src/lib.rs

```diff
@@ -16,21 +16,48 @@ use client_state::client_state_derive_impl;
 use consensus_state::consensus_state_derive_impl;
 use proc_macro::TokenStream as RawTokenStream;
 use syn::{parse_macro_input, DeriveInput};
+use utils::Imports;
 
-#[proc_macro_derive(ClientState, attributes(generics, mock))]
-pub fn client_state_macro_derive(input: RawTokenStream) -> RawTokenStream {
+#[proc_macro_derive(IbcClientState, attributes(validation, execution))]
+pub fn ibc_client_state_macro_derive(input: RawTokenStream) -> RawTokenStream {
     let ast: DeriveInput = parse_macro_input!(input);
 
-    let output = client_state_derive_impl(ast);
+    let imports = Imports::new_ibc();
+
+    let output = client_state_derive_impl(ast, &imports);
+
+    RawTokenStream::from(output)
+}
+
+#[proc_macro_derive(IbcCoreClientState, attributes(validation, execution))]
```

Comment:
I have added a comment about the possibility of not maintaining two duplicate derive macros here for client state and later for consensus state.

Reply:
https://github.com/cosmos/ibc-rs/pull/1037#discussion_r1447532259